ledger/src/shred.rs

```diff
@@ -726,6 +737,27 @@ pub mod layout {
         }
     }
 
+    #[allow(dead_code)]
+    pub(crate) fn get_chained_merkle_root(shred: &[u8]) -> Option<Hash> {
+        let offset = match get_shred_variant(shred).ok()? {
+            ShredVariant::LegacyCode | ShredVariant::LegacyData => None,
+            ShredVariant::MerkleCode {
+                proof_size,
+                chained: true,
+                resigned,
+            } => merkle::ShredCode::get_chained_merkle_root_offset(proof_size, resigned).ok(),
+            ShredVariant::MerkleData {
+                proof_size,
+                chained: true,
+                resigned,
+            } => merkle::ShredData::get_chained_merkle_root_offset(proof_size, resigned).ok(),
+            _ => None,
```

Comment:
`_ =>` will silently ignore new variants added to the enum.
Can you please use the explicit:
```rust
ShredVariant::MerkleCode { proof_size:_, chained: false, resigned:_ } => None,
ShredVariant::MerkleData { proof_size:_, chained: false, resigned:_ } => None,
```
so that we get compile errors when new variants are added?